File: Tech Prototype/SJTU_Stranding_backend/src/main/java/com/reins/sjtu_stranding/service/ChatService.java
package com.reins.sjtu_stranding.service;

import com.reins.sjtu_stranding.entity.Chat;
import com.reins.sjtu_stranding.entity.ChatInfo;

import java.util.List;


public interface ChatService {
    List<Chat> getChat(int userId);
    List<ChatInfo> getChatInfo(int chatId);
    List<ChatInfo> sendChat(int sender, int receiver, String dialog);
    void addChat(int sender, int receiver);
}

File: Tech Prototype/sjtu_stranding/src/services/chatService.js
import {postRequest} from "../utils/ajax";
export const getChat = (userId, callback) => {
    const data = {userId: userId};
    const url = "http://localhost:8081/getChat"
    postRequest(url, data, callback);
};
export const getChatInfo = (chatId, callback) => {
    const data = {chatId: chatId};
    const url = "http://localhost:8081/getChatInfo"
    postRequest(url, data, callback);
};
export const sendChat = (sender, receiver, dialog, callback) => {
    const data = {sender: sender, receiver:receiver, dialog:dialog};
    const url = "http://localhost:8081/sendChat"
    postRequest(url, data, callback);
};
export const addChat = (sender,receiver,callback) =>{
    const data = {sender: sender, receiver:receiver};
    const url = "http://localhost:8081/addChat"
    postRequest(url, data, callback);
}